hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/CastMap.java

```diff
@@ -165,21 +192,132 @@ void add(int pos, LogicalType fromType, LogicalType toType) {
         }
         break;
       }
+      case ARRAY: {
+        if (from == ARRAY) {
+          LogicalType fromElementType =  fromType.getChildren().get(0);
+          LogicalType toElementType = toType.getChildren().get(0);
+          return array -> doArrayConversion((ArrayData) array, fromElementType, toElementType);
+        }
+        break;
+      }
+      case MAP: {
+        if (from == MAP) {
+          return map -> doMapConversion((MapData) map, fromType, toType);
+        }
+        break;
+      }
+      case ROW: {
+        if (from == ROW) {
+          // Assumption: InternalSchemaManager should produce a cast that is of the same size
+          return row -> doRowConversion((RowData) row, fromType, toType);
+        }
+        break;
+      }
       default:
     }
-    return null;
+    throw new IllegalArgumentException(String.format("Unsupported conversion for %s => %s", fromType, toType));
```

Comment:
Do not throws `RuntimeException` in nested calling code path, it is very obstacle for the invoker to get the perception. Either throws a checked exception or return null as of before.

Reply:
Modified them to be checked exceptions, not sure if i did it correctly, please take a look, thank you.